src/instrumenter/transpiling_context.ts

```diff
@@ -194,35 +192,40 @@ export class TranspilingContext {
-            if (
-                this.instrSiteType === InstrumentationSiteType.FunctionAnnotation ||
-                this.instrSiteType === InstrumentationSiteType.StateVarUpdated
-            ) {
-                const uncheckedBlock = this.factory.makeUncheckedBlock([]);
-                this.uncheckedBlocks.push(uncheckedBlock);
-                containerBody.insertAtBeginning(uncheckedBlock);
-
-                this.oldMarkerStack = [[uncheckedBlock, "end"]];
+        /// TODO(dimo): This logic is convoluted and doesn't belong in the constructor. Instead perhaps we should have
+        /// helper methods like `initMarkersSinglePointWrapper()`, `initMarkersTwoPointWrapper`, `initMarkersSinglePointStatement`
+        /// and `initMarkersTwoPointStatement`
+        if (
+            this.instrSiteType === InstrumentationSiteType.SinglePointWrapper ||
+            this.instrSiteType === InstrumentationSiteType.TwoPointWrapper
+        ) {
+            if (gte(instrCtx.compilerVersion, "0.8.0")) {
+                const containerBody = this.containerFun.vBody as Block;
+                if (this.instrSiteType === InstrumentationSiteType.TwoPointWrapper) {
+                    const uncheckedBlock = this.factory.makeUncheckedBlock([]);
+                    this.uncheckedBlocks.push(uncheckedBlock);
+                    containerBody.insertAtBeginning(uncheckedBlock);
+
+                    this.oldMarkerStack = [[uncheckedBlock, "end"]];
+                }
+
+                const newUncheckedBlock = this.factory.makeUncheckedBlock([]);
+                this.uncheckedBlocks.push(newUncheckedBlock);
+                containerBody.appendChild(newUncheckedBlock);
+                this.newMarkerStack = [[newUncheckedBlock, "end"]];
+            } else {
+                const containerBody = this.containerFun.vBody as Block;
+                if (this.instrSiteType === InstrumentationSiteType.TwoPointWrapper) {
+                    const firstStmt = containerBody.vStatements[0];
+
+                    this.oldMarkerStack = [[containerBody, ["before", firstStmt]]];
+                }
+
+                this.newMarkerStack = [[containerBody, "end"]];
             }
-
-            const newUncheckedBlock = this.factory.makeUncheckedBlock([]);
-            this.uncheckedBlocks.push(newUncheckedBlock);
-            containerBody.appendChild(newUncheckedBlock);
-            this.newMarkerStack = [[newUncheckedBlock, "end"]];
         } else {
-            const containerBody = this.containerFun.vBody as Block;
-            if (
-                this.instrSiteType === InstrumentationSiteType.FunctionAnnotation ||
-                this.instrSiteType === InstrumentationSiteType.StateVarUpdated
-            ) {
-                const firstStmt = containerBody.vStatements[0];
-
-                this.oldMarkerStack = [[containerBody, ["before", firstStmt]]];
-            }
-
-            this.newMarkerStack = [[containerBody, "end"]];
```

Comment:
Should assert that this.instrSiteType === InstrumentationSiteType.Custom to guard against future new values in the enum

Reply:
Fixed in [a482f6b](https://github.com/ConsenSys/scribble/pull/154/commits/a482f6b39338dedcd43bdfe3f23939aed396ad48).